server/models/board.py

```diff
@@ -0,0 +1,156 @@
+
+                if not (r, c) in self.__edges:
+                    self.__edges[(r, c)] = {}
+
+                # add an un-owned edge from the current node to the node below and to the right
+                # do not add an edge that goes out of bounds (r + 1 exceeds rows, c + 1 exceeds cols)
+                if not c == self.cols - 1:
+                    self.__edges[(r, c)][(r, c + 1)] = None
+
+                if not r == self.rows - 1:
+                    self.__edges[(r, c)][(r + 1, c)] = None
+
+    '''
+    '' Returns the owner of the edge (p1, p2)
+    ''  p1:     A tuple describing source node (row, col)
+    ''  p2:     A tuple describing target node (row, col)
+    ''
+    ''  Returns the owner of the edge, or None if the edge is unowned
+    '''
+    def get_edge_owner(self, p1, p2):
+
+        if len(p1) != 2 or len(p2) != 2:
+            raise ValueError("A point was not a tuple of len 2: {}, {}".format(p1, p2))
+
+        if not self.__is_valid_edge(p1, p2):
+            raise ValueError("Edge is not valid:  {}, {}".format(str(p1), str(p2)))
+
+        p1, p2 = self.__coerce_points(p1, p2)
+
+        return self.__edges[p1][p2]
+
+    '''
+    '' Returns True if the edge (p1, p2) is owned
+    ''  p1:     A tuple describing source node (row, col)
+    ''  p2:     A tuple describing target node (row, col)
+    ''
+    ''  Returns the owner of the edge, or None if the edge is unowned
+    '''
+    def edge_is_owned(self, p1, p2):
+        return self.get_edge_owner(p1, p2) is not None
+
+    '''
+    '' Claims an unowned edge from p1 to p2
+    ''
+    ''  p1:     A tuple describing source node (row, col)
+    ''  p2:     A tuple describing target node (row, col)
+    ''  player: the owner of the new edge
+    ''
+    ''  This implementation coerces the source node to always be above or to the left of the destination node
+    ''    eg: if claim_edge((1, 1), (0, 1), PLAYER) is called, edges[(0, 1)][(1, 1)] is assigned to PLAYER
+    ''
+    ''  Raises an ValueError if the edge is already owned
+    ''  Raises a ValueError if the edge isn't between two adjacent nodes
+    '''
+    def claim_edge(self, p1, p2, owner):
+
+        if len(p1) != 2 or len(p2) != 2:
+            raise ValueError("A point was not a tuple of len 2: {}, {}".format(p1, p2))
+
+        if not owner:
+            raise ValueError("Owner should not be None")
+
+        if not self.__is_valid_edge(p1, p2):
+            raise ValueError("Edge is not valid:  {}, {}".format(str(p1), str(p2)))
+
+        p1, p2 = self.__coerce_points(p1, p2)
+        if p1 not in self.__edges:
+            self.__edges[p1] = {}
+
+        if self.__edges[p1][p2]:
+            raise
+
+        created_boxes = self.__get_new_boxes(p1, p2)
+
+        self.__edges[p1][p2] = owner
+
+        # TODO:  implement _get_new_boxes
+        # for box in created_boxes:
+        #     self.boxes[box] = owner
+
+    '''
+    '' Returns the list of boxes that would be made by adding a line
```

Comment:
I thought adding a line can only capture a single box, but grants you an immediate placement, rather than a cooldown?